libgnucash/engine/qofid.cpp

```diff
@@ -303,7 +303,7 @@ qof_collection_set_data (QofCollection *col, gpointer user_data)
 
 /* =============================================================== */
 
-struct _iterate
+struct __iterate
```

Comment:
I think we shouldn't use this name here: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl8-use-a-consistent-naming-style

Reply:
WTF?1 @bhardwajs is right. That's not just core guidelines, identifiers with double underscores are reserved *[in the standard](https://en.cppreference.com/w/cpp/language/identifiers) and using them is UB. So how the heck is UB going to improve LTO?

Reply:
Also [reserved by gcc](https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html).

Reply:
Not sure why he chose that as the rename. I assume the rationale for the patch is that there's a `struct _iterate` defined in both `qofid.cpp` and `qofbook.cpp`. I'll ask.

Reply:
Rebased with a different name.